src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/transformer/cas1/Cas1SpaceBookingRequirementsTransformer.kt

```diff
@@ -0,0 +1,23 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.cas1
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1SpaceBookingRequirements
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.Cas1SpaceCharacteristic
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.CharacteristicEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.PlacementRequirementsEntity
+
+@Component
+class Cas1SpaceBookingRequirementsTransformer {
+  fun transformJpaToApi(jpa: PlacementRequirementsEntity) = Cas1SpaceBookingRequirements(
+    apType = jpa.apType,
+    gender = jpa.gender,
+    essentialCharacteristics = jpa.essentialCriteria.mapNotNull { it.asCas1SpaceCharacteristic() },
+    desirableCharacteristics = jpa.desirableCriteria.mapNotNull { it.asCas1SpaceCharacteristic() },
+  )
+
+  private fun CharacteristicEntity.asCas1SpaceCharacteristic() = try {
+    Cas1SpaceCharacteristic.valueOf(this.propertyName!!)
+  } catch (_: Exception) {
+    null
```

Comment:
do we expect this to happen (e.g. are there characteristics in the database that don't map to JPA)? My main concern would be a subtle naming mismatch not being mapped. Would throwing an exception here be an issue?

Reply:
I've copied this logic from `PlacementRequestTransformer.characteristicToCriteria` which swallows the exception.

I think the main benefit would be in testing, where an arbitrary `CharacteristicEntity` won't cause a test to fail because of the exception being thrown.

Reply:
OK, we'll leave as is and can always return to this later